src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -487,7 +491,9 @@ event::ptr primitive_inst::realloc_if_needed() {
         }
     }
 
-    bool can_reuse_buffer = _outputs[0] && updated_layout.count() <= max_output_layout_size;
+    bool can_reuse_buffer = _outputs[0]
+                             && updated_layout.count() <= max_output_layout_size
+                             && !need_reset_output_memory();
```

Comment:
BTW I think we should limit this to (is_concat && need_reset_output_memory) because for other nodes like reduce or convolution, we will reuse memory b/w iterations, and enqueue mem reset. The only exception is concat for now. So please fix this point and add some comments.